worker/fanconfigurer/fanconfigurer.go

```diff
@@ -44,7 +44,10 @@ func (fc *FanConfigurer) processNewConfig() error {
 	defer fc.mu.Unlock()
 
 	fanConfig, err := fc.config.Facade.FanConfig()
-	if err != nil {
+	// NOTE(nvinuesa - 2024/05/27): This worker and this facade is being
+	// removed in 4.0, and therefore we check for NotImplemented errors
+	// and not fail in that case, to keep 3.6->4.0 migrations working.
+	if err != nil && !errors.Is(err, errors.NotImplemented) {
```

Comment:
I think we should log out that the implementation is not being used.

Reply:
Just added a trace, I'm not sure this is 100% useful though. The only case this appear is a migration to 4.0.